react/components/Button/index.tsx

```diff
@@ -0,0 +1,395 @@
+  block: PropTypes.bool,
+  /** Loading state */
+  isLoading: PropTypes.bool,
+  /** [DEPRECATED] If you are using just an Icon component inside, use this as true */
+  icon: PropTypes.bool,
+  /** @ignore For internal use
+   * Sets reduced paddings in order to keep the button squareish if it
+   * only has an icon  */
+  iconOnly: PropTypes.bool,
+  /** (Button spec attribute) */
+  id: PropTypes.string,
+  /** Data attribute */
+  testId: PropTypes.string,
+  /** (Button spec attribute) */
+  autoFocus: PropTypes.bool,
+  /** (Button spec attribute) */
+  autoComplete: PropTypes.string,
+  /** (Button spec attribute) */
+  disabled: PropTypes.bool,
+  /** @ignore Forwarded Ref */
+  forwardedRef: refShape,
+  /** (Button spec attribute) */
+  name: PropTypes.string,
+  /** (Button spec attribute) */
+  type: PropTypes.string.isRequired,
+  /** (Button spec attribute) */
+  value: PropTypes.string,
+  /** Label of the Button */
+  children: PropTypes.node.isRequired,
+  /** onClick event. */
+  onClick: PropTypes.func,
+  /** URL for link mode. Converts the button internally to a link. */
+  href: PropTypes.string,
+  /** onMouseEnter event */
+  onMouseEnter: PropTypes.func,
+  /** onMouseLeave event */
+  onMouseLeave: PropTypes.func,
+  /** onMouseOver event */
+  onMouseOver: PropTypes.func,
+  /** onMouseOut event */
+  onMouseOut: PropTypes.func,
+  /** onMouseUp event */
+  onMouseUp: PropTypes.func,
+  /** onMouseDown event */
+  onMouseDown: PropTypes.func,
+  /** onFocus event */
+  onFocus: PropTypes.func,
+  /** onBlur event */
+  onBlur: PropTypes.func,
+  /** @ignore deprecated
+   * Cancels out left padding */
+  collapseLeft: PropTypes.bool,
+  /** @ignore deprecated
+   * Cancels out right padding */
+  collapseRight: PropTypes.bool,
+  /** */
+  isGrouped: PropTypes.bool,
+  /** */
+  isFirstOfGroup: PropTypes.bool,
+  /** */
+  isLastOfGroup: PropTypes.bool,
+  /** */
+  isActiveOfGroup: PropTypes.bool,
+  /** Link spec */
+  target: PropTypes.string,
+  /** Link spec */
+  rel: PropTypes.string,
+  /** Link spec */
+  referrerPolicy: PropTypes.string,
+  /** Link spec */
+  download: PropTypes.string,
+  /** When terciary, the upper case can be prevented */
+  noUpperCase: PropTypes.bool,
+  /** Disables label wrapping */
+  noWrap: PropTypes.bool,
+}
+
+type Props = InferProps<typeof propTypes>
+
+const Button: React.FC<Props> = props => {
+  const {
+    id,
+    size,
+    block,
+    variation,
+    icon,
+    children,
+    isLoading,
+    collapseLeft,
+    collapseRight,
+    isGrouped,
+    isActiveOfGroup,
+    isFirstOfGroup,
+    isLastOfGroup,
+    href,
+    target,
+    rel,
+    referrerPolicy,
+    download,
+    noUpperCase,
+    noWrap,
+    testId,
+    autoFocus,
+    name,
+    onMouseEnter,
+    onMouseLeave,
+    onMouseOver,
+    onMouseOut,
+    onMouseUp,
+    onMouseDown,
+    onFocus,
+    onBlur,
+    forwardedRef,
+    type,
+    value,
+  } = props
+
+  React.useEffect(() => {
```

Comment:
Destructure { useEffect } from react.

Reply:
@matheusps the proposed idea is to transform what we already have in javascript to typescript, not to use React's new features. Hooks is not available for all versions that use the styleguide within the context (vtexio). So improvements to the component should be made in the future, unfortunately.

Reply:
Ow, okay. Got it.

Reply:
I thought it was to improve the component itself, my bad.